drone/src/agent/engines/docker/mod.rs

```diff
@@ -386,11 +386,15 @@ impl Engine for DockerInterface {
 
             Ok(EngineBackendStatus::Running { addr })
         } else {
-            match state.exit_code {
-                None => Ok(EngineBackendStatus::Terminated),
-                Some(0) => Ok(EngineBackendStatus::Exited),
-                Some(_) => Ok(EngineBackendStatus::Failed),
-            }
+            let status = match state.exit_code {
+                None => EngineBackendStatus::Terminated,
+                Some(0) => EngineBackendStatus::Exited,
+                Some(_) => EngineBackendStatus::Failed,
+            };
+
+            tracing::info!(?status, ?state.exit_code, "Set to terminal state.");
```

Comment:
I think we should log the entire `state`. There's quite a bit of useful info in there:
```
    pub status: Option<ContainerStateStatusEnum>,
    pub running: Option<bool>,
    pub paused: Option<bool>,
    pub restarting: Option<bool>,
    pub oom_killed: Option<bool>,
    pub dead: Option<bool>,
    pub pid: Option<i64>,
    pub exit_code: Option<i64>,
    pub error: Option<String>,
    pub started_at: Option<String>,
    pub finished_at: Option<String>,
    pub health: Option<Health>,
```